packages/shared/lib/core/wallet/actions/events-handlers/handleNewOutputEvent.ts

```diff
@@ -66,11 +68,19 @@ export async function handleNewOutputEventInternal(walletId: string, payload: Ne
 
     if (isAccountOutput) {
         const accounts = await wallet.accounts()
-        const depositAddress = await getDepositAddress(wallet)
-        updateSelectedWallet({
-            accountOutputs: accounts,
-            depositAddress,
-        })
+        if (accounts.length === 1) {
+            const accountId = (accounts[0]?.output as AccountOutput).accountId
```

Comment:
what does the SDK return when we call `wallet.accounts()`, is it all account outputs, or only block issuer accounts? 
I ask because if they return all, in order to be 100% accurate, we should check that this output has the block issuer feature enabled (for which we can make a util function to check if an account is a block issuer) otherwise we could potentially be setting as main account an account that cant issue blocks

Reply:
Yes, `accounts` returns all outputs, whether or not they have the issuer feature block or not. This is what the sdk function filter does
`Checks whether the output is a(n) AccountOutput`